### ISSUE TITLE
Cordova 9 fail install

### ISSUE THREAD

**Issue by videmort** [opentok/cordova-plugin-opentok#165]
Cordova plugin add --> 
Using "requireCordovaModule" to load non-cordova module "q" is not supported. Instead, add this module to your dependencies and use regular "require" to load it.

**Comment by videmort** [comment]
--> https://github.com/apache/cordova/issues/85

**Comment by mmubasher** [comment]
run `npm install` after running cordova plugin add command @videmort

**Comment by GuillermoLovotrico** [comment]
Same problem. Only for iOS
«Installing "cordova-plugin-opentok" for ios
Failed to install 'cordova-plugin-opentok': CordovaError: Using "requireCordovaModule" to load non-cordova module "q" is not supported. Instead, add this module to your dependencies and use regular "require" to load it."»

**Comment by tusharvikky** [comment]
Any update on this? Cannot even get it installed.

**Comment by mmubasher** [comment]
Have you installed npm modules before cordova build using `npm install`? also is your environment as per following 
<table><thead>
<tr>
<th>cordova-android Version</th>
<th>Supported Android API-Levels</th>
<th>Equivalent Android Version</th>
</tr>
</thead><tbody>
<tr>
<td>8.X.X</td>
<td>19 - 28</td>
<td>4.4 - 9.0.0</td>
</tr>
<tr>
<td>7.X.X</td>
<td>19 - 27</td>
<td>4.4 - 8.1</td>
</tr>
<tr>
<td>6.X.X</td>
<td>16 - 26</td>
<td>4.1 - 8.0.0</td>
</tr>
<tr>
<td>5.X.X</td>
<td>14 - 23</td>
<td>4.0 - 6.0.1</td>
</tr>
<tr>
<td>4.1.X</td>
<td>14 - 22</td>
<td>4.0 - 5.1</td>
</tr>
<tr>
<td>4.0.X</td>
<td>10 - 22</td>
<td>2.3.3 - 5.1</td>
</tr>
<tr>
<td>3.7.X</td>
<td>10 - 21</td>
<td>2.3.3 - 5.0.2</td>
</tr>
</tbody></table>
see https://cordova.apache.org/docs/en/latest/guide/platforms/android/

**Comment by tusharvikky** [comment]
Yes, I tried ionic repair and then installed the plugin. Any recommendation would be great!

**Comment by mmubasher** [comment]
Please mention your `ionic info` output in project directory @tusharvikky

**Comment by tusharvikky** [comment]
@mmubasher Here. I was able to get it running on Android. However, iOS is still hard to debug. All I see is a white screen.

`➜ ionic info

Ionic:

   Ionic CLI                     : 5.2.2 (/Users/tushardeo/.nvm/versions/node/v10.10.0/lib/node_modules/ionic)
   Ionic Framework               : @ionic/angular 4.4.2
   @angular-devkit/build-angular : 0.13.9
   @angular-devkit/schematics    : 7.3.9
   @angular/cli                  : 7.3.9
   @ionic/angular-toolkit        : 1.5.1

Cordova:

   Cordova CLI       : 9.0.0 (cordova-lib@9.0.1)
   Cordova Platforms : android 8.0.0, ios 5.0.1
   Cordova Plugins   : cordova-plugin-ionic-keyboard 2.1.3, cordova-plugin-ionic-webview 4.1.1, (and 16 other plugins)

Utility:

   cordova-res : 0.3.0 (update available: 0.6.0)
   native-run  : 0.2.4 (update available: 0.2.8)

System:

   Android SDK Tools : 26.1.1 (/Users/tushardeo/Library/Android/sdk)
   ios-deploy        : 1.9.4
   ios-sim           : 8.0.1
   NodeJS            : v10.10.0 (/Users/tushardeo/.nvm/versions/node/v10.10.0/bin/node)
   npm               : 6.4.1
   OS                : macOS Mojave
   Xcode             : Xcode 10.2.1 Build version 10E1001


  ─────────────────────────────────────

     Update available: 5.2.2 → 5.2.3
      Run npm i -g ionic to update

  ─────────────────────────────────────

`

**Comment by mmubasher** [comment]
have you tried `ionic cordova build ios` after removing folders `platform` and `plugins`

**Comment by tusharvikky** [comment]
Yes. I have also tried out.

![image](https://user-images.githubusercontent.com/1810991/62490167-49cc5600-b7e6-11e9-902c-7e10d1f58d30.png)

**Comment by levidavidmurray** [comment]
Only solution that worked for me was rolling Cordova back to version 8.1.2

`yarn global add cordova@8.1.2` or `npm install -g cordova@8.1.2`

**Comment by dopplerDistortion** [comment]
Pull request sent for Cordova 9 fixes
